libyul/optimiser/CommonSubexpressionEliminator.h

```diff
@@ -58,8 +58,21 @@ class CommonSubexpressionEliminator: public DataFlowAnalyzer
 	using ASTModifier::visit;
 	void visit(Expression& _e) override;
 
+	void assignValue(YulString _variable, Expression const* _value) override;
 private:
+	struct ExpressionHash {
+		uint64_t operator()(Expression const*) const;
+	};
```

Comment:
MIght make sense to move to ``BlockHasher.h`` instead - it's generally a good thing to have such a functor for those hashers.